### PR TITLE
Homework-3: database integration

### PR DESCRIPTION
**Pull request by usvintsitski**
### For database WITH public access
* Create PostgreSql database with public access in AWS
* Setup database structure and data with the help of `db/init.sql` file
* Clone repository and checkout to `hw-3/db-integration` branch
* Go to product-service folder
* Create `envConfig.json` file and fill it with values for database connection like it is done in `envConfig.example.json`
* REMOVE  COMPLETELY `resources` property from `serverlessConfiguration` in `serverless.ts` file
* REMOVE import of `vpcConfig` in `serverless.ts` file
* Run `yarn`
* Use `yarn serverless:deploy` to deploy backend for products.

### For database WITHOUT public access
* Create PostgreSql database without public access in AWS
* Setup database structure and data with the help of `db/init.sql` file
* Create VPC security group and assign it to the database
* Clone repository and checkout to `hw-3/db-integration` branch
* Go to product-service folder
* Create `envConfig.json` file and fill it with values for database connection like it is done in `envConfig.example.json`
* Create `vpcConfig.json` file and fill it with values for VPC configuration (security group id and subnet id) like it is done in `vpcConfig.example.json`
* Run `yarn`
* Use `yarn serverless:deploy` to deploy backend for products.

### LINK TO THE API - https://jfrp9feove.execute-api.us-east-1.amazonaws.com/dev
Get products - GET https://jfrp9feove.execute-api.us-east-1.amazonaws.com/dev/products
Get product by id - GET https://jfrp9feove.execute-api.us-east-1.amazonaws.com/dev/products/{id}
Create product - POST https://jfrp9feove.execute-api.us-east-1.amazonaws.com/dev/products

[SWAGGER documentation is here](https://jfrp9feove.execute-api.us-east-1.amazonaws.com/dev/swagger)

### Additional (optional) tasks
+1 - POST/products lambda functions returns error 400 status code if product data is invalid
+1 - All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)
+1 - All lambdas do console.log for each incoming requests and their arguments
+1 - Transaction based creation of product

### CODE REVIEW

File: product-service/src/functions/getProductsList/createHandler.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+		try {
+			getLogger().log(event)
+			return formatJSONResponse(await getProductRepository().getAll());
+		} catch (error) {
+			return formatJSONResponse({ message: "Unknown error" }, 500);
+		}
+	};
+
+	return getProductsList;
+}
```
Comment (Lockd) [inline product-service/src/functions/getProductsList/createHandler.ts:22]: Just a suggestion - code would be more readable if indentation is consistent through out all files

File: product-service/src/functions/swagger/swagger.json [hunk 2]
```diff
@@ -43,6 +43,39 @@
...
+            "application/json"
+          ],
+          "parameters": [
+            {
+              "name": "body",
+              "in": "body",
+              "description": "Product to create",
+              "required": true,
+              "schema": {
+                "$ref": "#/definitions/Product"
```
Comment (Lockd) [inline product-service/src/functions/swagger/swagger.json:64]: Normally schema should be different if you are not expecting id to be passed as param

Comment (Lockd) [reply product-service/src/functions/swagger/swagger.json:64]: Now I see that in order for add product enpoint to work properly id is indeed required. 
Wouldn't it be better to have that logic on a backend side of things? e.g. generate uuid when trying to create a product?

Comment (usvintsitski) [reply product-service/src/functions/swagger/swagger.json:64]: Consumers of this API need to use uuid and not some serial numbers, so I see no problems in this approach (currently, at least).

File: product-service/src/functions/getProductsList/createHandler.ts (hunk 1 above)
Comment (usvintsitski) [reply product-service/src/functions/getProductsList/createHandler.ts:22]: Thank you for noticing it. I have missed some files after configuration of indentation. Will fix this issue in the following tasks.

**Pull request merged**